src/display/text_layer.js

```diff
@@ -318,7 +321,13 @@ class TextLayerRenderTask {
           }
           parent.append(this._container);
         } else if (item.type === "endMarkedContent") {
-          this._container = this._container.parentNode;
+          const parent = this._container.parentNode;
+          if (!parent || this._container === this.#initialContainer) {
```

Comment:
Please note that I originally only added the `parent`-check, however that caused the `text`-tests to layout the text in an incorrect position.

This won't be a problem in the viewer, where an in-memory `DocumentFragment` is being used, however in the test-suite a regular `div` is being used (since that's officially supported) and that one has a parent element which thus necessitated these changes.

Reply:
I think I'd have prefer having the fix in the `evaluator`, because it means that `getTextContent` is still returning some buggy data.
But in the meantime, this fix is very simple and doesn't add that much complexity, when fixing `getTextContent` would likely add more complexity.

Reply:
> I think I'd have prefer having the fix in the `evaluator`, because it means that `getTextContent` is still returning some buggy data.

Completely agreed, but I'm thinking we can land this as-is such that it's included in the next release.

> But in the meantime, this fix is very simple and doesn't add that much complexity, when fixing `getTextContent` would likely add more complexity.

It might not be *that* difficult to address actually, I'll try to look into this a bit more later.

Reply:
I haven't run a single test locally, so I suppose that I might have broken *something*, however my naive idea looks like this: https://github.com/mozilla/pdf.js/compare/master...Snuffleupagus:pdf.js:issue-15629-2

Reply:
And it passes all `text`-tests locally, so might actually work out.

Reply:
However, I suppose that one argument against doing that is that the rendering code (in `src/display/canvas.js`) handles unbalanced operators just fine without tracking that in the `PartialEvaluator`.

Reply:
Yep we should do the same for the operatorList, but we can do that later.